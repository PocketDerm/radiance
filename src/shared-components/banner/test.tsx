import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Banner } from './index';

describe('Banner UI snapshots', () => {
  test('renders success type and text', () => {
    const { container } = render(
      <Banner content="Success Banner" type="success" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders error type and text', () => {
    const { container } = render(
      <Banner content="Error banner" type="error" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders info type and text', () => {
    const { container } = render(<Banner content="Default banner" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Banner with click handler', () => {
    jest.useFakeTimers();
    const spy = jest.fn();
    const { getByRole } = render(
      <Banner content="Banner with click handler" onClick={spy} />,
    );

    getByRole('button').click();

    jest.runAllTimers();
    expect(spy).toHaveBeenCalled();
  });
});
