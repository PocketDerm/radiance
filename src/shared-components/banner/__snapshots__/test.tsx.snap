// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner UI snapshots renders error type and text 1`] = `
.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #ffffff;
  font-size: 0.875rem;
}

.emotion-1 svg {
  height: 1rem;
  width: 1rem;
  fill: #ffffff;
}

.emotion-0 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #ffffff;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-3 {
  margin: -3px 0 0 1rem;
}

.emotion-7 {
  background: transparent;
  border: none;
  width: 100%;
  cursor: inherit;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #BD200F;
  box-shadow: 0px 8px 24px rgba(189,32,15,0.05);
}

.emotion-7:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #ffffff,0px 0px 0px 4px #332e54;
}

@media (min-width:768px) {
  .emotion-7 {
    margin: 0 auto 1rem;
  }
}

<button
  className="emotion-7 emotion-8"
  tabIndex={-1}
>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-1 emotion-2"
    >
      <svg
        className="emotion-0"
        height={16}
        width={16}
      />
    </div>
    <div
      className="emotion-3 emotion-4"
    >
      Error banner
    </div>
  </div>
</button>
`;

exports[`Banner UI snapshots renders info type and text 1`] = `
.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #ffffff;
  font-size: 0.875rem;
}

.emotion-1 svg {
  height: 1rem;
  width: 1rem;
  fill: #ffffff;
}

.emotion-0 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #ffffff;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-3 {
  margin: -3px 0 0 1rem;
}

.emotion-7 {
  background: transparent;
  border: none;
  width: 100%;
  cursor: inherit;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #332e54;
  box-shadow: 0px 8px 24px rgba(51,46,84,0.05);
}

.emotion-7:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #ffffff,0px 0px 0px 4px #332e54;
}

@media (min-width:768px) {
  .emotion-7 {
    margin: 0 auto 1rem;
  }
}

<button
  className="emotion-7 emotion-8"
  tabIndex={-1}
>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-1 emotion-2"
    >
      <svg
        className="emotion-0"
        height={16}
        width={16}
      />
    </div>
    <div
      className="emotion-3 emotion-4"
    >
      Default banner
    </div>
  </div>
</button>
`;

exports[`Banner UI snapshots renders success type and text 1`] = `
.emotion-7 {
  background: transparent;
  border: none;
  width: 100%;
  cursor: inherit;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #2B6E33;
  box-shadow: 0px 8px 24px rgba(43,110,51,0.05);
}

.emotion-7:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #ffffff,0px 0px 0px 4px #332e54;
}

@media (min-width:768px) {
  .emotion-7 {
    margin: 0 auto 1rem;
  }
}

.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #ffffff;
  font-size: 0.875rem;
}

.emotion-1 svg {
  height: 1rem;
  width: 1rem;
  fill: #ffffff;
}

.emotion-0 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #ffffff;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-3 {
  margin: -3px 0 0 1rem;
}

<button
  className="emotion-7 emotion-8"
  tabIndex={-1}
>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-1 emotion-2"
    >
      <svg
        className="emotion-0"
        height={16}
        width={16}
      />
    </div>
    <div
      className="emotion-3 emotion-4"
    >
      Success Banner
    </div>
  </div>
</button>
`;
